### ISSUE TITLE
Date.toInstant() not implemented - workaround

### ISSUE THREAD

**Issue by mungler** [google/j2objc#1843]
Hi,

We made some recent changes in our library which broke our iOS build. The error was similar to this:

```
Test.java:16: error: cannot find symbol
        Instant inst = date.toInstant();
                           ^
  symbol:   method toInstant()
  location: variable date of type java.util.Date
  ```

We had to replace this call:

```
Instant inst = date.toInstant();
```

with the equivalent workaround:

```
Instant inst = Instant.ofEpochMilli(date.getTime());
```

It would be lovely if this method could be implemented to save others from the same pitfall, but just in case, hopefully this workaround will help. 

Example problematic class attached:

[Test.java.txt](https://github.com/google/j2objc/files/8131786/Test.java.txt)

**Comment by tomball** [comment]
We hadn't included those methods (`toInstant()` and from`(Instant)`) because `java.util.Date` is in `jre_core` and the `jre_time` subset library adds 20M to the app (actually, most of that bloat is from `jre_icu`, which `java_time` depends upon). 

However, I think we can add those methods by moving them to an Objective-C category in `java.time.Instant`. That way, developers like yourself won't have to do any build or runtime magic to get them added, as that will happen when `java.time.Instant` is loaded. Sound reasonable?

**Comment by mungler** [comment]
It does indeed, thanks Tom!

**Comment by tomball** [comment]
Fixed in [37e5e93b83aa39d7a9a81c01d65bca738c0a972b](https://github.com/google/j2objc/commit/37e5e93b83aa39d7a9a81c01d65bca738c0a972b). The magic is in [Instant.java](https://github.com/google/j2objc/blob/master/jre_emul/android/platform/libcore/ojluni/src/main/java/java/time/Instant.java#L1367), if anyone is interested.
